woocommerce/woocommerce
# Shipping Settings: Highlight text on focus of cost input values

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





There was some confusion about default values, especially `0.00` looking like placeholders in the Shipping modals. This PR makes it such that input text is selected on focus. I'm hoping this will reduce the confusion.

This way, typing after focus will avoid inputs looking like `00.00100` when a merchant types `100`.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new shipping zone , `/wp-admin/admin.php?page=wc-settings&tab=shipping&zone_id=new`
2. Try adding any method.
3. Find a cost input and note selection of the value when focus is placed on the input.
4. Save and continue. Try the same when editing the configured method and be sure to try different types of methods. The same behaviour should result.
5. Lastly, does the behaviour make sense? Is it intuitive?



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Select text when focusing cost inputs on Shipping Settings modals.

#### Comment 

</details>
